desktop/desktop
# Surface errors from branch creation to user

## Overview



**Closes #8306**
**Closes #5997**
**Closes #8106**

## Description

#8306 alerted us to the fact that we weren't propagating errors from Git to the user when creating a branch. This PR let's any Git errors bubble up from the `createBranch` method where they can be caught by the `performFailableOperation` in `AppStore._createBranch`.

https://github.com/desktop/desktop/blob/84c5928b06f979147be397b0a19b8ec320754160/app/src/lib/stores/app-store.ts#L2815-L2817

This is the only callsite for the `createBranch` method and as such I deem this to be a very low-risk fix.

I think there's a bigger picture solution here where we should consider whether to show a warning if the user is attempting to create a branch that only differs in case (it's quite possible that the branch can still be created, but it's a recipe for trouble) but given that this bug has been around for at least 2 years and we haven't gotten all that many reports I think this is a good enough solution for now.

### Testing notes

1. Create a brand new repository (not strictly necessary but avoids the risk of packed-refs throwing a wrench in the works)
2. Create a branch named `bananas`
3. Attempt to create a branch named `BaNaNaS`

Expected result: An error dialog informing you that the branch exists
Actual result (prior to this PR): infinitely spinning create branch dialog

## Release notes



Notes: `[Fixed] Show the underlying error if branch creation fails.`
